les/api.go

```diff
@@ -410,15 +410,15 @@ func (api *LightClientAPI) RequestPeerGatewayFees() error {
 		cost := peerNode.GetRequestCost(GetGatewayFeeMsg, int(1))
 		err := peerNode.RequestGatewayFee(genReqID(), cost)
 		if err != nil {
-			continue
+			return err
 		}
 	}
 	return nil
 }
 
 // SuggestGatewayFee suggests the best light server to choose based on different factors. Currently only minPeerGatewayFee.
-func (api *LightClientAPI) SuggestGatewayFee() (map[string]*GatewayFeeInformation, error) {
-	bestGatewayFeeInfo, err := api.le.handler.MinPeerGatewayFee()
+func (api *LightClientAPI) SuggestGatewayFee() (*FullGatewayFeeInformation, error) {
```

Comment:
You don't need to return the node ID from this method. Instead it should just return `GatewayFeeInformation` and  `FullGatewayFeeInformation` can be removed.

Reply:
okay I'll remove that struct